types/liftoff/index.d.ts

```diff
@@ -17,65 +17,63 @@ declare class Liftoff extends EventEmitter {
+        listener: (name: string, module: unknown) => void
     ): this;
-    prependOnceListener(event: 'requireFail', listener: (name: string, err: any) => void): this;
+    prependOnceListener(event: 'loader:failure' | 'preload:failure' | 'preload:before', listener: (name: string, err: Error) => void): this;
     prependOnceListener(
         event: 'respawn',
         listener: (flags: string[], child: NodeJS.Process) => void
     ): this;
     removeListener(
-        event: 'require',
-        listener: (name: string, module: ExtensionDescriptor) => void
+        event: 'loader:success' | 'preload:success',
+        listener: (name: string, module: unknown) => void
     ): this;
-    removeListener(event: 'requireFail', listener: (name: string, err: any) => void): this;
+    removeListener(event: 'loader:failure' | 'preload:failure' | 'preload:before', listener: (name: string, err: Error) => void): this;
     removeListener(
         event: 'respawn',
         listener: (flags: string[], child: NodeJS.Process) => void
     ): this;
-    off(event: 'require', listener: (name: string, module: ExtensionDescriptor) => void): this;
-    off(event: 'requireFail', listener: (name: string, err: any) => void): this;
+    off(event: 'loader:success' | 'preload:success', listener: (name: string, module: unknown) => void): this;
+    off(event: 'loader:failure' | 'preload:failure' | 'preload:before', listener: (name: string, err: Error) => void): this;
     off(event: 'respawn', listener: (flags: string[], child: NodeJS.Process) => void): this;
-    removeAllListeners(event?: 'require' | 'requireFail' | 'respawn'): this;
-    listeners(event: 'require' | 'requireFail' | 'respawn'): Function[]; // tslint:disable-line:ban-types
-    rawListeners(event: 'require' | 'requireFail' | 'respawn'): Function[]; // tslint:disable-line:ban-types
-    emit(event: 'require', name: string, module: ExtensionDescriptor): boolean;
-    emit(event: 'requireFail', name: string, err: any): boolean;
+    removeAllListeners(event?: 'preload:success' | 'preload:failure' | 'preload:before' | 'respawn'): this;
+    listeners(event: 'preload:success' | 'preload:failure' | 'preload:before' | 'respawn'): Function[]; // tslint:disable-line:ban-types
+    rawListeners(event: 'preload:success' | 'preload:failure' | 'preload:before' | 'respawn'): Function[]; // tslint:disable-line:ban-types
+    emit(event: 'preload:success', name: string, module: unknown): boolean;
+    emit(event: 'preload:failure' | 'preload:before', name: string, err: Error): boolean;
     emit(event: 'respawn', flags: string[], child: NodeJS.Process): boolean;
-    eventNames(): Array<'require' | 'requireFail' | 'respawn'>;
-    listenerCount(type: 'require' | 'requireFail' | 'respawn'): number;
+    eventNames(): Array<'preload:success' | 'preload:failure' | 'preload:before' | 'respawn'>;
+    listenerCount(type: 'preload:success' | 'preload:failure' | 'preload:before' | 'respawn'): number;
```

Comment:
Ok, I've tried out something new to reduce repetition and add uniformity to definitions. You can replace this whole block with the following definition:
```ts
    addListener<TEvent extends keyof Liftoff.Events>(event: TEvent, listener: Liftoff.Events[TEvent]): this;
    addListener(event: string | symbol, listener: (...args: any[]) => void): this;
    on<TEvent extends keyof Liftoff.Events>(event: TEvent, listener: Liftoff.Events[TEvent]): this;
    on(event: string | symbol, listener: (...args: any[]) => void): this;
    once<TEvent extends keyof Liftoff.Events>(event: TEvent, listener: Liftoff.Events[TEvent]): this;
    once(event: string | symbol, listener: (...args: any[]) => void): this;
    prependListener<TEvent extends keyof Liftoff.Events>(event: TEvent, listener: Liftoff.Events[TEvent]): this;
    prependListener(event: string | symbol, listener: (...args: any[]) => void): this;
    prependOnceListener<TEvent extends keyof Liftoff.Events>(event: TEvent, listener: Liftoff.Events[TEvent]): this;
    prependOnceListener(event: string | symbol, listener: (...args: any[]) => void): this;
    removeListener<TEvent extends keyof Liftoff.Events>(event: TEvent, listener: Liftoff.Events[TEvent]): this;
    removeListener(event: string | symbol, listener: (...args: any[]) => void): this;
    off<TEvent extends keyof Liftoff.Events>(event: TEvent, listener: Liftoff.Events[TEvent]): this;
    off(event: string | symbol, listener: (...args: any[]) => void): this;
    listeners<TEvent extends keyof Liftoff.Events>(event: TEvent): Array<Liftoff.Events[TEvent]>;
    listeners(event: string | symbol): Array<(...args: any[]) => void>;
    rawListeners<TEvent extends keyof Liftoff.Events>(event: TEvent): Array<Liftoff.Events[TEvent]>;
    rawListeners(event: string | symbol): Array<(...args: any[]) => void>;
    emit<TEvent extends keyof Liftoff.Events>(event: TEvent, ...args: Parameters<Liftoff.Events[TEvent]>): boolean;
    emit(event: string | symbol, ...args: any[]): boolean;
```

You'll have to additionally add the new `Events` type to the `Liftoff` namespace below:
```ts
    interface Events {
        'loader:success': (name: string, module: unknown) => void;
        'preload:success': (name: string, module: unknown) => void;
        'loader:failure': (name: string, err: Error) => void;
        'preload:failure': (name: string, err: Error) => void;
        'preload:before': (name: string) => void;
        respawn: (flags: string[], child: NodeJS.Process) => void;
    }
```